core/chains/evm/client/rpc_client_test.go

```diff
@@ -209,6 +209,16 @@ func TestRPCClient_SubscribeFilterLogs(t *testing.T) {
 	lggr := logger.Test(t)
 	ctx, cancel := context.WithTimeout(tests.Context(t), tests.WaitTimeout(t))
 	defer cancel()
+	t.Run("Failed SubscribeFilterLogs when WSURL is empty", func(t *testing.T) {
+		// ws is optional when LogBroadcaster is disabled, however SubscribeFilterLogs will return error if ws is missing
+		httpURL := url.URL{}
+		observedLggr, _ := logger.TestObserved(t, zap.DebugLevel)
+		rpcClient := client.NewRPCClient(observedLggr, url.URL{}, &httpURL, "rpc", 1, chainId, commonclient.Primary, 0, commonclient.QueryTimeout, commonclient.QueryTimeout, "")
```

Comment:
Could `httpURL` be omitted?
```suggestion
		observedLggr, _ := logger.TestObserved(t, zap.DebugLevel)
		rpcClient := client.NewRPCClient(observedLggr, url.URL{}, nil, "rpc", 1, chainId, commonclient.Primary, 0, commonclient.QueryTimeout, commonclient.QueryTimeout, "")
```

Reply:
For this specific test it can't, since Dial() will fail due to `errWSAndHTTPBothMissing`. I can add another one test that covers when both WS and HTTPURL are empty